spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -322,6 +322,12 @@ public void onSuccess(SendResult<byte[], byte[]> result) {
 		props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ByteArrayDeserializer.class);
 		props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
 		props.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, 100);
+		
+		props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, anonymous ? "latest" : "earliest");
+		if (!ObjectUtils.isEmpty(consumerProperties.getExtension().getStartOffset())) {
```

Comment:
I think we need to move this setting to end of the method (or after `configuration` properties are set) to give the highest precedence to `binder` level consumer properties over Spring Boot Kafka properties